asmcomp/cmmgen.ml

```diff
@@ -2181,6 +2188,27 @@ and transl_prim_1 env p arg dbg =
   | Pnegbint bi ->
       box_int dbg bi
         (Cop(Csubi, [Cconst_int 0; transl_unbox_int dbg env bi arg], dbg))
+  | Pclzbint bi -> begin
+      match (transl_unbox_int dbg env bi arg) with
+      | Cconst_natint 0n ->
+        let n = (match bi with
+          | Pnativeint -> 8*size_int
+          | Pint32 -> 32
+          | Pint64 -> 64)
+        in tag_int (Cconst_int n) dbg
+      | _ ->
+        let res = Cop((Cclz false),
+                      [make_unsigned_int bi
+                         (transl_unbox_int dbg env bi arg) dbg], dbg) in
+        if bi = Pint32 && size_int = 8 then
```

Comment:
I think endianess is handled in individual target emit.mlp. 
By the way, I tested it in one big endian configuration, for s390 and passes the tests. Big endian ppc machine in precheck seems to be down.  
I haven't prepared the tests with endianness in mind, because this is purely in register operation and does not involve moves between memory and registers. I will add some tests for big endian and try it on ppc target with qemu as well.